File: it/structures/js/default_naming-default/upper_snake.js
export class Value {
  constructor(foo_bar) {
    this.foo_bar = foo_bar;
  }

  static decode(data) {
    let v_foo_bar = data["FOO_BAR"];

    if (v_foo_bar === null || v_foo_bar === undefined) {
      throw new Error("FOO_BAR" + ": required field");
    }

    if (typeof v_foo_bar !== "string") {
      throw Error("expected string");
    }

    return new Value(v_foo_bar);
  }

  encode() {
    const data = {};

    if (this.foo_bar === null || this.foo_bar === undefined) {
      throw new Error("FOO_BAR: is a required field");
    }

    data["FOO_BAR"] = this.foo_bar;

    return data;
  }
}

File: it/structures/python2/default_naming-default/upper_snake.py
class Value:
  def __init__(self, foo_bar):
    self.__foo_bar = foo_bar

  @property
  def foo_bar(self):
    return self.__foo_bar

  @foo_bar.setter
  def foo_bar(self, foo_bar):
    self.__foo_bar = foo_bar

  @staticmethod
  def decode(data):
    f_foo_bar = data["FOO_BAR"]

    if not isinstance(f_foo_bar, unicode):
      raise Exception("not a string")

    return Value(f_foo_bar)

  def encode(self):
    data = dict()

    if self.foo_bar is None:
      raise Exception("missing required field: foo_bar")

    data["FOO_BAR"] = self.foo_bar

    return data

  def __repr__(self):
    return "<Value foo_bar:{!r}>".format(self.foo_bar)
